getting-started/installation-windows-msys2.rst

```diff
@@ -0,0 +1,290 @@
+
+Again: you have a single installation of MSYS2, these different shells are just setups to use different compilers. Installation and update of your MSYS2 system only has to be done once - you can use any of the shell options for that.
+
+Update MSYS2 with following command::
+
+    $ pacman -Syu
+  
+After this finishes (let it close the bash shell), open bash again and run the same command again to finish the updates. The same procedure is used for regular updates of the MSYS2 installation. An up-to-date system shows::
+
+    $ pacman -Syu
+    :: Synchronizing package databases...
+     mingw32 is up to date
+     mingw64 is up to date
+     msys is up to date
+    :: Starting core system upgrade...
+     there is nothing to do
+    :: Starting full system upgrade...
+     there is nothing to do
+
+Install the necessary tools (perl is already part of the base system)::
+
+    $ pacman -S tar make
+
+Packages with such "simple" names are part of the MSYS2 environment and work for all compilers/toolchains that you may install on top on MSYS2.
+
+Install compiler toolchains
+---------------------------
+Packages that are part of a MinGW toolchain start with the prefix "mingw-w64-x86_64-" for the 64bit toolchain or "mingw-w64-i686-" for the 32bit toolchain.
+(The "w64" part identifies the host system will be different when using a 32bit MSYS2 environment, e.g. on a 32bit Windows host.)
+
+Install the MinGW 32bit and/or MinGW 64bit toolchains::
+
+    $ pacman -S mingw-w64-x86_64-toolchain
+    $ pacman -S mingw-w64-i686-toolchain
+    
+Each complete toolchain needs about 900MB of disk space.
+If you want to cut down the needed disk space (to about 50%), instead of hitting return when asked which packages of the group to install, you can select the minimal set of packages required for compiling EPICS Base: ``binutils``, ``gcc`` and ``gcc-libs``.
+
+If you are not sure, check your set of tools is complete and everything is installed properly::
+
+    $ pacman -Q
+    ...
+    make 4.3-1
+    perl 5.32.0-2
+    mingw-w64-x86_64-...
+    mingw-w64-i686-...
+    ...
+
+Update your installation regularly
+----------------------------------
+As mentioned above, you can update your complete installation (including all tools and compiler toolchains) at any time using::
+
+    $ pacman -Syu
+
+You should do this in regular intervals.
+
+Download and build EPICS Base
+-----------------------------
+
+::
+
+    $ cd $HOME
+    $ wget https://epics-controls.org/download/base/base-7.0.4.1.tar.gz
+    $ tar -xvf base-7.0.4.1.tar.gz
+    $ cd base-R7.0.4.1
+    $ export EPICS_HOST_ARCH=windows-x64-mingw
+    $ make
+
+When using the MinGW 32bit toolchain, the "MSYS MinGW 32-bit" shell must be used and EPICS_HOST ARCH must be set to "win32-x86-mingw".
+
+During the compilation, there will probably be warnings, but there should be no error. You can choose any EPICS Base version to build, the procedure remains the same.
```

Comment:
Can we specify build time ? like 30min to 60min depending on the system. When I first compiled EPICS I waited staring at screen for an hour expecting it to finished in 10-15 minutes.

Reply:
That depends on a lot of factors. Have a look at a recent AppVeyor build of EPICS 7 (https://ci.appveyor.com/project/epics-base-7/epics-base/builds/36440444) and you can see that on those machines, it's between 14 and 29 minutes, the 3.15 branch (https://ci.appveyor.com/project/epics-base-3/epics-base/builds/36371542) builds in 6 to 10 minutes.

These are relatively small two-core virtual machines. If you use a slow file system or a single core or a slow machine (or a combination), it can easily be an hour. On a faster machine (8 or 16 cores), it will be faster.

Reply:
Point taken, though. Will add a sentence about build times.